docs/lesson-plans/week-4.md

```diff
@@ -0,0 +1,623 @@
+    - Importance of a Clear Server/Client Interface and 
+    - Hosting and Localhost
+- HTTP/REST Basics
+    - What is HTTP
+    - Status Codes Overview
+    - Request Types
+    - JSON
+- Backend Basics and GET Requests
+    - Idea of Routing on the Backend
+        - How to Make a New Route
+        - Static Routing
+        - Route Params
+    - Using Postman to Hit Up APIs
+    
+## Basic Plan
+1. Servers
+    - What is a server?
+    - What is a client?
+    - Importance of communication
+    - Hosting
+        - IP Addresses
+        - Localhost
+        - Ports
+        - Ngrok
+2. HTTP Requests and REST
+    - What is HTTP
+        - Status Codes in HTTP
+            - Informational Responses
+            - Successful Responses
+            - Redirect Responses
+            - Client Errors
+            - Server Errors
+        - JSONs
+            - Values
+            - Arrays
+            - Objects
+        - Request types overview
+            - GET
+    - What is REST (REpresentational State Transfer)
+    - Postman and API clients
+3. Routing on the Backend
+    - Structure of the Backend
+    - What is routing?
+    - Routing with Vert.x
+        - Static routing
+        - Route parameters
+        - Query parameters
+        - Creating a Sub-Router
+    - Processor Interfaces
+    
+## Servers
+
+The server-client architecture is the way that the web works. In a simple explanation, servers are programs which
+await requests sent by client programs (like web browsers), and the requests are processed so that a response can 
+be returned.
+
+![Server-client architecture model](./img/server-client-model.png)
+
+*Gnome-fs-client.svg: David Vignoni Gnome-fs-server.svg: David Vignoni derivative work: Calimo 
+/ LGPL (http://www.gnu.org/licenses/lgpl.html)*
+
+### What is a server?
+
+Servers are the programs which power every website and remote service you use. In a somewhat simplified way, they
+are usually just programs that are always-on and listening whose purpose is to run remote operations 
+for you. They handle incoming and outgoing requests, errors, and running functionality you design! They can be run for 
+programs that you use locally as well as remote. 
+
+As an example, when you play video games with other people, that's spinning up a server that handles running everyone's 
+game and the interactions that may happen between each player and AI.
+
+### What is a client?
+
+A client is a program that makes requests to servers. Servers can end up having client functionality if they need to 
+interact with other servers, but in a lot of cases, they can just be standalone programs too.
+
+From the example of the game above, the client side of that would be the part of the game that you interact with and 
+download locally, which ends up sending requests to the (usually remote) server. Another example of a client could be 
+your web browser, which sends requests when you try to load websites or send information.
```

Comment:
I think we can mention how the client is closely tied to the UI/UX to help with the explanation!

Reply:
let me know if what I put is good.